subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/BuildExperimentRunner.java

```diff
@@ -16,216 +16,14 @@
 
 package org.gradle.performance.fixture;
 
-import org.gradle.api.Action;
-import org.gradle.internal.concurrent.CompositeStoppable;
-import org.gradle.performance.measure.MeasuredOperation;
 import org.gradle.performance.results.MeasuredOperationList;
-import org.gradle.util.GFileUtils;
 
-import java.io.File;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.concurrent.atomic.AtomicBoolean;
+public interface BuildExperimentRunner {
```

Comment:
Is this a general "build experiment" runner, or is this only used for performance tests? Looks like it's a generic thing but the only two implementations I see are for performance tests.

We should either call everything a build experiment or a performance test I think.

Reply:
The current nomenclature in the performance project are:
- Scenario, e.g. `non-abi change on largeJavaMultiProject`
- Build Experiment, e.g. getting the measurements for Gradle 5.7 on a scenario
- Invocation, e.g running Gradle 5.7 on a scenario once

So `BuildExperimentRunner` is a perfectly good name in that context, since it runs one build experiment. Note that it does not run the whole performance scenario, nor conducts contains the test assertions. It also lives in the `org.gradle.performance` package, so I don't see a need to add `Performance` to it.

Reply:
I don't know why but I apparently imagined that subclasses had `PerformanceTest` in their names, but obviously not. Sorry about that, please ignore the noise.